sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -26,43 +26,64 @@ import org.apache.spark.sql.expressions.Aggregator
 // This file defines internal implementations for aggregators.
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+class TypedSumDouble[IN](val f: IN => Double)
+  extends Aggregator[IN, java.lang.Double, java.lang.Double] {
+
+  override def zero: java.lang.Double = 0.0
+  override def reduce(b: java.lang.Double, a: IN): java.lang.Double =
```

Comment:
I'm -1 on this change, it introduces a lot of overhead because of boxing. Can we revert it back and use `toColumnJava`?

Reply:
As discussed  previously the boxing is needed to have appropriate return types for min/max. This of course would not be needed if we align it to the current (incorrect) return values.
 
I have bounced back and forth between the return values multiple times now, so it might be worthwhile to have some more discussion.